src/fft.jl

```diff
@@ -1,63 +1,121 @@
 import FFTW
 
+# returns the lengths of the bounding rectangle in reciprocal space
+# that encloses the sphere of radius Gmax
+function bounding_rectangle(lattice::AbstractMatrix{T}, Gmax; tol=1e-8) where {T}
+    # If |B G| ≤ Gmax, then
+    # |Gi| ≤ |e_i^T B^-1 B G| ≤ |B^-T e_i| Gmax = |A_i| Gmax
+    # with B the reciprocal lattice matrix and A_i the i-th column of the lattice matrix
+    Glims_ = [norm(lattice[:, i]) / 2T(π) * Gmax for i in 1:3]
+
+    # Round up, unless exactly zero (in which case keep it zero in
+    # order to just have one G vector for 1D or 2D systems)
+    Glims = [0, 0, 0]
+    for i = 1:3
+        if Glims_[i] != 0
+            Glims[i] = ceil(Int, Glims_[i] .- tol)
+        end
+    end
+    Glims
+end
+
+_smallprimes() = [2, 3, 5, 7]
+
 @doc raw"""
 Determine the minimal grid size for the cubic basis set to be able to
 represent product of orbitals (with the default `supersampling=2`).
 
 Optionally optimize the grid afterwards for the FFT procedure by
 ensuring factorization into small primes.
 
-The function will determine the smallest cube containing the wave vectors
+The function will determine the smallest parallelepiped containing the wave vectors
  ``|G|^2/2 \leq E_\text{cut} ⋅ \text{supersampling}^2``.
 For an exact representation of the density resulting from wave functions
 represented in the spherical basis sets, `supersampling` should be at least `2`.
 """
-function determine_grid_size(lattice::AbstractMatrix{T}, Ecut; supersampling=2, tol=1e-8,
-                             ensure_smallprimes=true) where T
-    cutoff_Gsq = 2 * supersampling^2 * Ecut
-    Gmax = [norm(lattice[:, i]) / 2T(π) * sqrt(cutoff_Gsq) for i in 1:3]
+function determine_fft_size(lattice::AbstractMatrix{T}, Ecut; supersampling=2, tol=1e-8,
+                            ensure_smallprimes=true) where T
+    Gmax = supersampling * sqrt(2 * Ecut)
+    Glims = bounding_rectangle(lattice, Gmax; tol=tol)
 
-    # Round up, unless exactly zero (in which case keep it zero in
-    # order to just have one G vector for 1D or 2D systems)
-    for i = 1:3
-        if Gmax[i] != 0
-            Gmax[i] = ceil.(Int, Gmax[i] .- tol)
+    fft_size = Vec3(2 .* Glims .+ 1)
+    # Optimize FFT grid size: Make sure the number factorises in small primes only
+    if ensure_smallprimes
+        fft_size = nextprod.(Ref(_smallprimes()), fft_size)
+    end
+    fft_size
+end
+function determine_fft_size(model::Model, Ecut; kwargs...)
+    determine_fft_size(model.lattice, Ecut; kwargs...)
+end
+function lattice_diameter(latt)
```

Comment:
writing it out as `lattice` certainly does not hurt. Also some empty lines  around the functions please

Reply:
Also why not just call the function `diameter`?

Reply:
Problem with `lattice` is that it increases the confusion between lattice and recip_lattice, but OK